druid/examples/game_of_life.rs

```diff
@@ -210,6 +119,76 @@ impl Grid {
     }
 }
 
+#[derive(Clone, Copy)]
+struct GridPos {
+    row: usize,
+    col: usize,
+}
+
+impl GridPos {
+    pub fn above(self) -> Option<GridPos> {
+        if self.row == 0 {
+            None
+        } else {
+            Some(GridPos {
+                row: self.row - 1,
+                col: self.col,
+            })
+        }
+    }
+    pub fn below(self) -> Option<GridPos> {
+        if self.row == GRID_SIZE - 1 {
+            None
+        } else {
+            Some(GridPos {
+                row: self.row + 1,
+                col: self.col,
+            })
+        }
+    }
+    pub fn left(self) -> Option<GridPos> {
+        if self.col == 0 {
+            None
+        } else {
+            Some(GridPos {
+                row: self.row,
+                col: self.col - 1,
+            })
+        }
+    }
+    pub fn right(self) -> Option<GridPos> {
+        if self.col == GRID_SIZE - 1 {
+            None
+        } else {
+            Some(GridPos {
+                row: self.row,
+                col: self.col + 1,
+            })
+        }
+    }
+}
+
+#[derive(Clone)]
+struct ColorScheme {
+    colors: Vec<Color>,
+}
+
+#[derive(Clone, Lens, Data)]
+struct AppData {
+    grid: Grid,
+    drawing: bool,
+    paused: bool,
+    ups: f64,
```

Comment:
I would prefer some other name, I don't think it is obvious what `ups` means.

Reply:
`updates_per_second`?

Reply:
sure!